tests/test-extensions/heisenberg-extension/src/main/java/org/mule/test/heisenberg/extension/HeisenbergSource.java

```diff
@@ -173,15 +173,15 @@ public void onStart(SourceCallback<String, Object> sourceCallback) throws MuleEx
         context.fireOnHandle(NEXT_BATCH, TypedValue.of(frequency));
         sourceCallback.handle(result, context);
       }
-    }, 0, frequency, MILLISECONDS);
+    }, 0, 4000, MILLISECONDS);
```

Comment:
the frequency should still be the one from the config. It you need to change the frequency for a single test, change it in that test xml. If you need it to be generic, update the default value directly